api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1780,16 +1789,14 @@ type AzurePlatformSpec struct {
 	// +optional
 	MachineIdentityID string `json:"machineIdentityID,omitempty"`
 
-	// SecurityGroupName is the name of an existing security group on SubnetName. This field is provided as part of the
+	// SecurityGroupID is the ID of an existing security group on the SubnetID. This field is provided as part of the
 	// configuration for the Azure cloud provider, aka Azure cloud controller manager (CCM).
 	//
-	// Example: if your Network Security Group ID is /subscriptions/<subscriptionID>/resourcegroups/<resourceGroupName>/providers/Microsoft.Network/networkSecurityGroups/<securityGroupName>,
-	//          your SecurityGroupName is <securityGroupName>
-	//
-	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="SecurityGroupName is immutable"
-	// +optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="SecurityGroupID is immutable"
+	// +kubebuilder:validation:Required
+	// +required
 	// +immutable
-	SecurityGroupName string `json:"securityGroupName,omitempty"`
+	SecurityGroupID string `json:"securityGroupID,omitempty"`
```

Comment:
Is there some relation and requirement (including potential pre-configuration requirement) between SecurityGroupID and the VNetID? If so, we should document it

Reply:
There is a relation between this and the SubnetID. That relation is already documented `SecurityGroupID is the ID of an existing security group on the SubnetID.`.

There is a relation between the SubnetID and the VNetID/VNetName. That relation is already documented `This
	// subnet ID is expected to be a subnet within the VnetID / VnetName.`